imagebuildah/build.go

```diff
@@ -675,42 +693,304 @@ func (b *Executor) Execute(ib *imagebuilder.Builder, node *parser.Node) error {
 		if i < len(node.Children)-1 {
 			requiresStart = ib.RequiresStart(&parser.Node{Children: node.Children[i+1:]})
 		}
-		err := ib.Run(step, b, requiresStart)
-		if err != nil {
-			return errors.Wrapf(err, "error building at step %+v", *step)
+
+		if !b.layers && !b.noCache {
+			err := ib.Run(step, b, requiresStart)
+			if err != nil {
+				return errors.Wrapf(err, "error building at step %+v", *step)
+			}
+			continue
+		}
+
+		var (
+			cacheID string
+			err     error
+			imgID   string
+		)
+		// checkForCache will be true if b.cache is true and a cached intermediate image is found.
```

Comment:
b.layers?
Should checkForCache now be useLayers?

Reply:
Changed to `checkForLayers`.